tests/accessor/accessor_exceptions.hpp

```diff
@@ -0,0 +1,395 @@
+          int Dimension, sycl_stub::target Target, typename GetAccFunctorT>
+void check_exception(GetAccFunctorT construct_acc) {
+  auto queue = util::get_cts_object::queue();
+  DataT some_data(expected_val);
+  auto r = util::get_cts_object::range<Dimension>::get(1, 1, 1);
+  {
+    sycl::buffer<DataT, Dimension> data_buf(&some_data, r);
+
+    if constexpr (AccType == accessor_type::generic_accessor) {
+      auto action = [&] {
+        queue.submit([&](sycl::handler& cgh) { construct_acc(cgh, data_buf); })
+            .wait();
+      };
+      CHECK_THROWS_MATCHES(
+          action, sycl::exception,
+          sycl_cts::util::equals_exception(sycl::errc::invalid));
+    } else if constexpr (AccType ==
+                         accessor_tests_common::accessor_type::host_accessor) {
+      auto action = [&] {
+        queue.submit([&](sycl::handler& cgh) { construct_acc(cgh, data_buf); })
+            .wait();
+      };
+      CHECK_THROWS_MATCHES(
+          action, sycl::exception,
+          sycl_cts::util::equals_exception(sycl::errc::invalid));
+    } else if constexpr (AccType ==
+                         accessor_tests_common::accessor_type::local_accessor) {
+      auto action = [&] { construct_acc(queue); };
+
+      CHECK_THROWS_MATCHES(
+          action, sycl::exception,
+          sycl_cts::util::equals_exception(sycl::errc::kernel_argument));
+    }
+  }
+}
+
+/**
+ * @brief Provides functor that lets verify that local_accessor really thrown
+ *        exception.
+ * @tparam DataT Current data type
+ * @tparam Dimension Dimension size
+ * @tparam Target Current target
+ * @tparam AccType Current type of the accessor: generic_accessor,
+ *         local_accessor, or host_accessor
+ * @param type_name Current data type string representation
+ * @param access_mode_name Current access mode string representation
+ * @param target_name Current target string representation
+ */
+template <typename DataT, int Dimension, sycl_stub::target Target,
+          accessor_tests_common::accessor_type AccType>
+void test_exception_for_local_acc(const std::string& type_name,
+                                  const std::string& access_mode_name,
+                                  const std::string& target_name) {
+  auto section_name = get_section_name<Dimension>(
+      type_name, access_mode_name, target_name,
+      "Expecting exception when attempting to interact with local_accessor on "
+      "device side.");
+  SECTION(section_name) {
+    auto construct_acc = [](sycl::queue& queue) {
+      // Use a variable to avoid device code optimisation.
+      auto is_empty =
+          usm_helper::allocate_usm_memory<sycl::usm::alloc::shared, bool>(
+              queue);
+      queue
+          .submit([&](sycl::handler& cgh) {
+            auto is_empty_ptr = is_empty.get();
+            sycl_stub::local_accessor<DataT, Dimension> local_acc(cgh);
+            cgh.single_task<Kernel<AccType, DataT, Dimension, Target>>(
```

Comment:
Yes, and this test expected that invoking a `local_accessor` via `single_task` leads to a synchronous exception with the errc::kernel_argument error code.

Reply:
Actually, it's not so obvious from the first glance. We don't have a `SECTION` name here with details as we have for cases below. Vasily, would you mind to provide a comment with details on what specific error condition do you trigger? It might help to avoid similar confusions in future :)